delfin/drivers/dell_emc/vnx_block/alert_handler.py

```diff
@@ -0,0 +1,148 @@
+        self.navi_handler = navi_handler
+
+    @staticmethod
+    def parse_alert(alert):
+        try:
+            alert_model = dict()
+            # These information are sourced from device registration info
+            alert_model['alert_id'] = alert.get(AlertHandler.OID_MESSAGECODE)
+            alert_model['alert_name'] = alert.get(AlertHandler.OID_DETAILS)
+            alert_model['severity'] = AlertHandler.TRAP_LEVEL_MAP.get(
+                alert.get(AlertHandler.OID_SEVERITY),
+                constants.Severity.INFORMATIONAL)
+            alert_model['category'] = constants.Category.EVENT
+            alert_model['type'] = constants.EventType.EQUIPMENT_ALARM
+            alert_model['occur_time'] = time.time() * units.k
+            alert_model['description'] = alert.get(AlertHandler.OID_DETAILS)
+            alert_model['resource_type'] = constants.DEFAULT_RESOURCE_TYPE
+
+            return alert_model
+
+        except Exception as e:
+            LOG.error(e)
+            msg = (_("Failed to build alert model as some attributes missing "
+                     "in alert message."))
+            raise exception.InvalidResults(msg)
+
+    def handle_alters(self, alertlist):
+        alert_list = []
+        for alertinfo in alertlist:
+            alert_model = {
+                'alert_id': alertinfo.get('event_code'),
+                'alert_name': alertinfo.get('message'),
+                'severity': self.SEVERITY_MAP.get(
+                    alertinfo.get('event_code')[0:2]),
+                'category': constants.Category.EVENT,
+                'type': constants.EventType.EQUIPMENT_ALARM,
+                'occur_time': alertinfo.get('log_time_stamp'),
+                'description': alertinfo.get('message'),
+                'resource_type': constants.DEFAULT_RESOURCE_TYPE
+            }
+            alert_list.append(alert_model)
+        return alert_list
+
+    def list_alerts(self, query_para):
+        try:
+            alert_lists = []
+            host_ip = self.navi_handler.navi_host
+            # Get list of EmcVnxStor alerts
+            domains = self.navi_handler.get_domain()
+            if domains:
+                for domain in domains:
+                    host_ip = domain.get('ip_address')
+                    reslist = self.navi_handler.get_log(host_ip, query_para)
+                    alert_list = self.handle_alters(reslist)
+                    if alert_list:
+                        alert_lists.extend(alert_list)
+            else:
+                reslist = self.navi_handler.get_log(host_ip, query_para)
+                alert_list = self.handle_alters(reslist)
+                if alert_list:
+                    alert_lists.extend(alert_list)
+            if alert_lists:
+                alert_lists = self.arrange_alerts(alert_lists)
+            return alert_lists
+        except Exception as e:
+            err_msg = "Get alert failed: %s" % (six.text_type(e))
```

Comment:
This is too generic and also, it is not get alert but list alert. What if there is some other exception